sentry-graphql/src/main/java/io/sentry/graphql/ExceptionReporter.java

```diff
@@ -0,0 +1,146 @@
+
+  public void captureThrowable(
+      final @NotNull Throwable throwable,
+      final @NotNull ExceptionDetails exceptionDetails,
+      final @Nullable ExecutionResult result) {
+    final @NotNull IHub hub = exceptionDetails.getHub();
+    final Mechanism mechanism = new Mechanism();
+    mechanism.setType(MECHANISM_TYPE);
+    mechanism.setHandled(false);
+    final Throwable mechanismException =
+        new ExceptionMechanismException(mechanism, throwable, Thread.currentThread());
+    final SentryEvent event = new SentryEvent(mechanismException);
+    event.setLevel(SentryLevel.FATAL);
+
+    final Hint hint = new Hint();
+    setRequestDetailsOnEvent(hub, exceptionDetails, event);
+
+    if (result != null) {
+      @NotNull Response response = new Response();
+      Map<String, Object> responseBody = result.toSpecification();
+      response.setData(responseBody);
+      event.getContexts().setResponse(response);
+    }
+
+    hub.captureEvent(event, hint);
+  }
+
+  private void setRequestDetailsOnEvent(
+      final @NotNull IHub hub,
+      final @NotNull ExceptionDetails exceptionDetails,
+      final @NotNull SentryEvent event) {
+    hub.configureScope(
+        (scope) -> {
+          final @Nullable Request scopeRequest = scope.getRequest();
+          if (scopeRequest != null) {
+            setDetailsOnRequest(hub, exceptionDetails, scopeRequest);
+            event.setRequest(scopeRequest);
+          } else {
+            Request newRequest = new Request();
+            setDetailsOnRequest(hub, exceptionDetails, newRequest);
+            event.setRequest(newRequest);
+          }
+        });
+  }
+
+  private void setDetailsOnRequest(
+      final @NotNull IHub hub,
+      final @NotNull ExceptionDetails exceptionDetails,
+      final @NotNull Request request) {
+    request.setApiTarget("graphql");
+
+    if (exceptionDetails.isSubscription() || !isSpring) {
+      final @NotNull Map<String, Object> data = new HashMap<>();
+
+      data.put("data", exceptionDetails.getQuery());
+
+      if (hub.getOptions().isSendDefaultPii()) {
+        data.put("variables", exceptionDetails.getVariables());
+      }
+
+      // for Spring this will be replaced by RequestBodyExtractingEventProcessor
+      request.setData(data);
+    }
+  }
+
+  public static final class ExceptionDetails {
+
+    private final @NotNull IHub hub;
+    private final @Nullable InstrumentationExecutionParameters instrumentationExecutionParameters;
+    private final @Nullable DataFetchingEnvironment dataFetchingEnvironment;
+
+    private final boolean isSubscription;
+
+    public ExceptionDetails(
+        final @NotNull IHub hub,
+        final @Nullable InstrumentationExecutionParameters instrumentationExecutionParameters) {
+      this.hub = hub;
+      this.instrumentationExecutionParameters = instrumentationExecutionParameters;
+      dataFetchingEnvironment = null;
+      isSubscription = false;
```

Comment:
`isSubscription` is already false here.

Reply:
Yes because it's not a subscription at the moment if it uses `InstrumentationExecutionParameters`, only if coming in via `DataFetchingEnvironment`. I can make this more explicit and pass it into the ctor.